readthedocs/core/utils/__init__.py

```diff
@@ -75,23 +68,20 @@ def prepare_build(
         default_version = project.get_default_version()
         version = project.versions.get(slug=default_version)
 
+    build = Build.objects.create(
+        project=project,
+        version=version,
+        type='html',
+        state=BUILD_STATE_TRIGGERED,
+        success=True,
+        commit=commit
+    )
+
     kwargs = {
-        'record': record,
-        'force': force,
```

Comment:
I removed these two.

- `record` was removed from a the top level (when triggering a task) because there is no use case for that. We _always_ want to record a `Build` object in our database if there is a build happening. However, I kept it at `BuildCommand` level, because there are commands that we want to execute in the environment without showing them to the user. In that case, we are using `record=False`
- `force` stopped making sense when we started using `RTD_CLEAN_AFTER_BUILD=True`. With this setting, we are always "forcing a build" since there is no cached environment to re-use.

I'm happy to bring these features back (including the "cache whole build environment as `.tar.gz` on S3") when we have decided to start using them. I didn't want to refactor/maintain all that code that it's not currently used because it added a lot of extra work. I think that re-add them when required will be easier and cleaner with the new structure.

Reply:
This makes sense, I agree with your interpretation. I was originally looking for the BuildCommand record, but was just confused by the same argument here.